pkg/skaffold/schema/defaults/defaults.go

```diff
@@ -210,13 +210,13 @@ func setDefaultClusterTimeout(cluster *latest.ClusterDetails) error {
 }
 
 func setDefaultClusterPullSecret(cluster *latest.ClusterDetails) error {
-	cluster.PullSecretName = valueOrDefault(cluster.PullSecretName, constants.DefaultKanikoSecretName)
 	if cluster.PullSecret != "" {
```

Comment:
```suggestion
	if cluster.PullSecret != ""  && cluster.PullSecretName != "" {
```

Reply:
Also add tests

Reply:
```cluster.PullSecretName != "" ```
this could be skipped by user when user wants pullsecret to be created as constants.DefaultKanikoSecretName i.e kaniko-secret.
observe the below line
```cluster.PullSecretName = valueOrDefault(cluster.PullSecretName, constants.DefaultKanikoSecretName)```